src/main/query_profiler.cpp

```diff
@@ -464,6 +473,64 @@ static string RenderTiming(double timing) {
 	return timing_s + "s";
 }
 
+string QueryProfiler::ToHTML() const {
+	std::stringstream str;
+	HTMLToStream(str);
+	return str.str();
+}
+
+void QueryProfiler::HTMLToStream(std::ostream &ss) const {
```

Comment:
Can we avoid this duplication and only insert the ExplainFormat, then create a TreeRenderer from it in the original code?